src/android/FileTransferBackground.java

```diff
@@ -401,27 +410,32 @@ private void acknowledgeEvent(String eventId, CallbackContext context) {
      * Handle ACK data and send it to the JS.
      */
     private void handleAck(final Data ackData) {
-        // If upload was successful
         if (!ackData.getBoolean(UploadTask.KEY_OUTPUT_IS_ERROR, false)) {
-            // Read response from file if present
+            sendError(
+                ackData.getString(UploadTask.KEY_OUTPUT_ID),
+                ackData.getString(UploadTask.KEY_OUTPUT_FAILURE_REASON),
+                ackData.getBoolean(UploadTask.KEY_OUTPUT_FAILURE_CANCELED, false)
+            );
+        } else {
             String response = null;
             if (ackData.getString(UploadTask.KEY_OUTPUT_RESPONSE_FILE) != null) {
                 response = readFileToStringNoThrow(ackData.getString(UploadTask.KEY_OUTPUT_RESPONSE_FILE));
             }
 
-            sendSuccess(
-                    ackData.getString(UploadTask.KEY_OUTPUT_ID),
-                    response,
-                    ackData.getInt(UploadTask.KEY_OUTPUT_STATUS_CODE, -1 /* If this is sent, something is really wrong */)
-            );
-
-        } else {
-            // The upload was a failure
-            sendError(
-                    ackData.getString(UploadTask.KEY_OUTPUT_ID),
-                    ackData.getString(UploadTask.KEY_OUTPUT_FAILURE_REASON),
-                    ackData.getBoolean(UploadTask.KEY_OUTPUT_FAILURE_CANCELED, false)
-            );
+            long startUploadTime = ackData.getLong("output_upload_start_time", 0);
+            long finishUploadTime = ackData.getLong("output_upload_finish_time", 0);
+            long uploadDuration = finishUploadTime - startUploadTime;
+
+            HashMap<String, Object> uploadData = new HashMap<>();
+            uploadData.put("outputId", ackData.getString(UploadTask.KEY_OUTPUT_ID));
+            uploadData.put("response", response);
+            uploadData.put("statusCode", ackData.getInt(UploadTask.KEY_OUTPUT_STATUS_CODE, -1));
+            
+            if (uploadDuration > 0) {
```

Comment:
Because of the `if`, will we not get an error in method `sendSuccess()` when doing:
```java
long uploadDuration = (long) uploadData.get("uploadDuration");
long finishUploadTime = (long) uploadData.get("finishUploadTime");
```
Even if it's `0`, we can send `0` directly right?